engines/private/private.cpp

```diff
@@ -0,0 +1,1086 @@
+        _mixer->stopHandle(_bgSoundHandle);
+    } else {
+        _mixer->stopHandle(_fgSoundHandle);
+    }
+}
+
+void PrivateEngine::loadImage(const Common::String &name, int x, int y) {
+    debugC(1, kPrivateDebugFunction, "%s(%s,%d,%d)", __FUNCTION__, name.c_str(), x, y);
+    Common::File file;
+    Common::String path = convertPath(name);
+    if (!file.open(path))
+        error("unable to load image %s", path.c_str());
+
+    _image->loadStream(file);
+    Graphics::Surface *surf = _image->getSurface()->convertTo(_pixelFormat, _image->getPalette());
+    _compositeSurface->transBlitFrom(*surf, *_origin + Common::Point(x,y), _transparentColor);
+    surf->free();
+    delete surf;
+    _image->destroy();
+    //drawScreen();
+}
+
+void PrivateEngine::drawScreenFrame(Graphics::Surface *surf) {
+    surf->copyRectToSurface(*_frame, 0, 0, Common::Rect(0, 0, _screenW, _screenH));
+}
+
+
+Graphics::ManagedSurface *PrivateEngine::loadMask(const Common::String &name, int x, int y, bool drawn) {
+    debugC(1, kPrivateDebugFunction, "%s(%s,%d,%d,%d)", __FUNCTION__, name.c_str(), x, y, drawn);
+    Common::File file;
+    Common::String path = convertPath(name);
+    if (!file.open(path))
+        error("unable to load mask %s", path.c_str());
+
+    _image->loadStream(file);
+    Graphics::ManagedSurface *surf = new Graphics::ManagedSurface();
+    surf->create(_screenW, _screenH, _pixelFormat);
+    surf->fillRect(screenRect, _transparentColor);
+    Graphics::Surface *csurf = _image->getSurface()->convertTo(_pixelFormat, _image->getPalette());
+    surf->transBlitFrom(*csurf, Common::Point(x,y));
+    csurf->free();
+    delete csurf;
+    _image->destroy();
+
+    if (drawn) {
+        drawMask(surf);
+    }
+
+    return surf;
+}
+
+void PrivateEngine::drawMask(Graphics::ManagedSurface *surf) {
+    _compositeSurface->transBlitFrom(surf->rawSurface(), *_origin, _transparentColor);
+    //drawScreen();
+}
+
+void PrivateEngine::drawScreen() {
+    Graphics::ManagedSurface *surface = _compositeSurface;
+
+    if (_videoDecoder) {
+        const Graphics::Surface *frame = _videoDecoder->decodeNextFrame();
+        //frame->create(_videoDecoder->getWidth(), _videoDecoder->getHeight(), _pixelFormat);
+        //frame->copyFrom(*_videoDecoder->decodeNextFrame());
+        Graphics::Surface *cframe = frame->convertTo(_pixelFormat, _videoDecoder->getPalette());
+        Common::Point center((_screenW - _videoDecoder->getWidth())/2, (_screenH - _videoDecoder->getHeight())/2);
+        surface->transBlitFrom(*cframe, center);
+        //frame->free();
+        cframe->free();
+        //delete frame;
+        delete cframe;
+    }
+
+    Graphics::Surface *screen = g_system->lockScreen();
+    if (_mode == 1) {
+        drawScreenFrame(screen);
+    }
+    Common::Rect window(_origin->x, _origin->y, _screenW - _origin->x, _screenH - _origin->y);
+    screen->copyRectToSurface(*surface, _origin->x, _origin->y, window);
```

Comment:
Why not use `g_system->copyRectToScreen()` instead of locking the screen and copying to it like this?

Reply:
Fixed. **Please** review the new screen related code.